tests/test_restfulapi.py

```diff
@@ -140,3 +145,67 @@ def test_known_values_always_have_their_types_converted(self):
             )
         finally:
             os.environ.pop("APPTEST_FOO", None)
+
+
+class FetchChangeUnitTest(unittest.TestCase):
+    def setUp(self):
+        self.request = make_request()
+        self.config = testing.setUp()
+        self.config.add_route("documents", pattern="/documents/{document_id}")
+
+    def make_documents(self, quant):
+        for i in range(quant):
+            self.request.matchdict = {"document_id": f"0000-0000-23-24-223{i}"}
+            self.request.validated = apptesting.document_registry_data_fixture()
+            restfulapi.put_document(self.request)
+
+    def test_fetch_changes(self):
+        self.assertEqual(
+            restfulapi.fetch_changes(self.request),
+            {"since": "", "limit": 500, "results": []},
+        )
+
+    def test_limit_must_be_int(self):
+        self.request.GET["limit"] = "foo"
+        self.assertRaises(HTTPBadRequest, restfulapi.fetch_changes, self.request)
+
+    def test_limit_return_correct_value(self):
```

Comment:
@jamilatta, este teste não está fazendo sentido, já que o request é simulado. Está validando um recurso do Pyramid usado para teste, não a aplicação em si.

Reply:
@patymori, na realidade o que está sendo testado é que o valor passado pelo argumento _limit_ esteja sendo retornado na estrutura `{"since": "", "limit": 1000, "results": []}`